### PR TITLE
week 3 assessment completed

### PR DESCRIPTION
**Pull request by sayoung0820**
# Submitting Week 3 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [✅] Coding Challenges
- [✅ ] Interview Questions
- [✅] React Debugging

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -20,8 +20,35 @@ const fibLength1 = 6
...
+
+const fibSequence = (number) => {
+  if(number === 6) {
+    return[1, 1, 2, 3, 5, 8]
+  } else if(number === 10) {
+    return[1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
+  }
+}
+
+
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:51]: Shannon this is what we call "hard coded" in development. Your syntax is awesome, the logic however only works for 2 situations. This Fib sequence function should be able to work for ANY possible positive number greater than 2.  

It should do so by processing the logic necessary to figure out what the solution would be **dynamically**. 

I recommend comparing your notes against what we review in class for this problem and making sure that next weeks assessment doe not hard code your solutions.

File: code-challenges.test.js [hunk 2]
```diff
@@ -47,10 +74,76 @@ const studyMinutesWeek2 = {
...
+    })
+}
+
+const sortMinutes2 = (Object) => {
+  Object.keys(studyMinutesWeek2).sort().reduce(
+    (obj, key) => {
+      obj[key] = unordered[key]
+      return obj
+    })
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:127]: You are on the right track with your solution here. but what you should be accessing is the values and not the keys, its a bit tricksy. 

Remember that a function is reusable code, so at most we only want a single function per solution.

File: code-challenges.test.js [hunk 3]
```diff
@@ -64,4 +157,61 @@ const accountTransactions2 = [250, -89, 100, -96]
...
+  return array1.map((sum => value => sum += value)(0))
+}
+
+const accTran2 = (array2) => {
+  return array2.map((sum => value => sum += value)(0))
+}
+
+const accTran3 = (array3) => {
+  return array3.map((sum => value => sum += value)(0))
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:213]: Again this should all be a single function that works dynamically. But good job making the tests pass!

File: interview-questions.md [hunk 4]
```diff
@@ -8,48 +8,66 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 Researched answer:
+Yarn is a package manager for coding that allows you to use and share code with other developers. Yarn let's you see solutions to problems that other developers have already resolved. 
 
 3. What is an anonymous function in JavaScript?
 
 Your answer:
+An anonymous function is an unamed function in JavaScript. They can be used inside other functions and also passed as arguments to other functions.
 
 Researched answer:
+Anonymous functions are functions that do not have any names associated with them. They can have multiple arguments, making them essential for reducing the number of code files and simplifying programs.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:30]: First sentence here is spot on!

File: interview-questions.md [hunk 5]
```diff
@@ -8,48 +8,66 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+Anonymous functions are functions that do not have any names associated with them. They can have multiple arguments, making them essential for reducing the number of code files and simplifying programs. 
+
 
 4. What is the difference between state values and props in React?
 
 Your answer:
+The main difference between state values and props is that props are read only meaning that they are not able to be changed. 
 
 Researched answer:
+Props and State values both hold information that affects the output, however; props are passed to the component while state values are managed within the component. Props are values received from another source, while state values are internal and controlled by the component itself.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:39]: Love this answer!

File: react-debugging.html [hunk 6]
```diff
@@ -65,11 +65,12 @@
         // No need to change anything ABOVE this line ^
 
         return (
+          <>
```
Comment (WalkerTexas-coder) [inline react-debugging.html:68]: Looks like this fragment is missing a buddy

File: react-debugging.html [hunk 7]
```diff
@@ -65,11 +65,12 @@
...
           <h1>Draw a Card</h1>
-          <button onClick={}>
+          <button onClick={drawCard}>
             Click to Draw a Card
           </button>
-          <button onclick={shuffle}>
+          <button onClick={shuffle}>
```
Comment (WalkerTexas-coder) [inline react-debugging.html:73]: Great work here!

File: interview-questions.md (hunk 5 above)
Comment (WalkerTexas-coder) [inline interview-questions.md:39]: Excellent!

**Review by WalkerTexas-coder — APPROVED** [review]
Shannon you are knocking your syntax outta the park! Great work. I see you put a lot of hard work into this assessment. Your pseudo code is clean and your solutions are on their way to being awesome. That said this assessment has a few points that bring it into the novice level. 

Your logic for problem solving needs to be focused on taking in and transforming data / making evaluations in response to the data. I have more notes below on this point. Your interview answers are almost 100% correct but could use a bit more exploration of the topics to notch it up to the next level. I would love to hear about times you used the tools in question. Looks like you were 90% on the way with the react concepts but just missed the last touch of the react fragment. If you can bring the code challenges into alignment with the goals of writing dynamic functions that solve for test cases and edge cases you'll be well on your way to advanced beginner and competent.
